test/r_tree/r_tree_test.dart

```diff
@@ -13,7 +13,7 @@ main() {
         RTreeDatum<String> item = new RTreeDatum<String>(new Rectangle(0, 0, 1, 1), 'Item 1');
 
         tree.insert(item);
-        Iterable<RTreeDatum<String>> items = tree.search(item.rect);
+        var items = tree.search(item.rect);
```

Comment:
These are all changed to `var` to let the inference figure out the type. The return type of search is `Iterable<RTreeDatum<dynamic>>` so specifying it as `Iterable<RTreeDatum<String>>` was tightening down the type and causing runtime type errors.